recipes/conda-project/meta.yaml

```
@@ -0,0 +1,56 @@
+{% set name = "conda-project" %}
+{% set version = "0.1.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/conda-incubator/conda-project/archive/refs/tags/{{ version }}.tar.gz
+  sha256: 6707d4d0aa0b0a12d4baa63ede878a7a391c9302a7a6ee916253d1966ea435a3
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  noarch: python
+  entry_points:
+    - conda-project = conda_project.cli.main:main
+
+requirements:
+  host:
+    - python >=3.8, <3.12
+    - pip
+    - setuptools
```

According to https://github.com/conda-incubator/conda-project/blob/0.1.0/pyproject.toml#L2 
```suggestion
    - setuptools
    - setuptools-scm
```

---

👍

Out of curiosity, why do we have to list these explicitly if `pip` will grab them when it processes the install backend?

---

pip is not allowed to install anything from pypi during processing, as that would be a supply chain risk. Everything to be defined via conda available dependencies and sources with sha256.

You strictly don't need setuptools-scm anymore as the switch to pypi is packaging an sdist with burned in version... that afaik makes setuptools-scm an optional requirement.

You need setuptools-scm + such a file https://github.com/conda/conda/blob/main/.git_archival.txt to be able to package github release tarballs (it allows setuptools-scm to calculate the right version from the archive during build).

---

Got it. Thanks for the detailed explanation!

I tried the sit archive support in setuptools-scm but didn't have any luck getting it working. The PyPI solution works well though, so I gave up on that yak shave 🙂. Maybe I'll come back to it another time.